modules/reindex/src/test/java/org/elasticsearch/index/reindex/RoundTripTests.java

```diff
@@ -120,9 +121,24 @@ public void testBulkIndexByScrollResponse() throws IOException {
         assertResponseEquals(response, tripped);
     }
 
+    public void testRethrottleRequest() throws IOException {
+        RethrottleRequest request = new RethrottleRequest();
+        request.setRequestsPerSecond((float) randomDoubleBetween(0, Double.POSITIVE_INFINITY, false));
```

Comment:
Wouldn't this potentially be larger than Float.MAX_VALUE, be cast to a float and end up negative, causing an exception? (it could be I don't know what happens in Java numerics for this too)